### PR TITLE
Fix AWS client

### PR DESCRIPTION
**Pull request by shiimaxx**
Fix issue that region config for AWS client was hardcode to `ap-northeast-1`. And some refactor.

Fix
- Region config fetches by EC2 metadata and that set when creating AWS client.

Refactor
- Create AWS client at initializing daemon, and pass to the handler with [service injection](https://github.com/go-martini/martini/tree/v1.0#service-injection) (to avoid redundant code by error handling in each handle functions).

### CODE REVIEW

File: command/daemon.go [hunk 1]
```diff
@@ -129,10 +129,18 @@ func CmdDaemon(c *cli.Context) {
 
 	isAutoScalingNode := c.Bool("enable-autoscaling-node")
 	if isAutoScalingNode {
-		client := autoscaling.NewNodeAWSClient()
+		nodeClient, err := autoscaling.NewNodeAWSClient()
+		if err == nil {
+			m.Map(nodeClient)
```
Comment (netmarkjp) [inline command/daemon.go]: I cannot find document about `Map`.

It looks that `nodeClient` will map to handler automatically = pass to handler as argument (Type: `*NodeAWSClient` ), is it right?

**Comment by netmarkjp** [comment]
I don't know much about implementation of `aws-sdk-go`, but typical problem of this situation(change from one-time client to long-life shared client) is below.

- mixing multi-thead(multi-goroutine) : may cause mixed-result
- fail recovery : may cause situtation `restart the process to fix problem`

Do you know whether or not `aws-sdk-go` is safe from these problems ?

File: command/daemon.go (hunk 1 above)
Comment (shiimaxx) [reply command/daemon.go]: That's right. I implemented it with see [this section](https://github.com/go-martini/martini#services).  
Services set with `Map` is mapping with argument of same type when invoke handler and will use.  

Also `Map` is method of `type injector` that is part of [codegangsta/inject](https://github.com/codegangsta/inject/blob/HEAD/inject.go#L139-L144)` and it embedded in [Matini](https://github.com/go-martini/martini/blob/v1.0/martini.go#L31)

**Comment by shiimaxx** [comment]
@netmarkjp 
Thanks.

- mixing multi-thead(multi-goroutine) : may cause mixed-result

Safe to this issue, because AWS SDK for Go's service clients is safe to use concurrently.

https://github.com/aws/aws-sdk-go/tree/v1.13.39#how-to-use-the-sdks-aws-service-clients
> These clients are safe to use concurrently.

- fail recovery : may cause situtation restart the process to fix problem

I couldn't find documents that mention this issue.  
However, I concerned because many places in AWS SDK for Go invokes `panic()`.

It may be good simplify implement (create service client in handler instead service client inject to the handler).  
What do you think?

**Comment by netmarkjp** [comment]
@shiimaxx Thanks for response. It looks good :+1:

**Pull request merged**